### ISSUE TITLE
Feature request: Snoozing messages

### ISSUE THREAD

**Issue by Fell** [mjl-/mox#225]
Some mail providers allow you to "snooze" a message for a given time. The message will be removed from your Inbox, and re-appear as an unread message at the time you specified. 

Before switching to Mox, I used a script similar to [Dovecot Snooze](https://www.wogri.at/scripts/dovecot-snooze/) for this purpose, which ran as a simple cronjob.

I think the implementation is simple, yet effective. A special `Snooze` folder is created, and with it a set of subfolders like `Snooze.Until Friday 18:00`. At the mentioned time, all messages within that folder will be moved to `Inbox` and marked as unread. Mail clients will generally trigger a new message notification when this happens.

Discussion questions:

- Are other users interested in such a feature?
- How could this be implemented in Mox?
  - Hidden implementation: The folders are created by the user and Mox does its best to parse the folder labels and tries to act accordingly.
  - Using Admin UI: Times can be configured in the Admin UI and folders are managed automatically.
  - Static implementation: The feature can only be enabled/disabled and only a hardcoded set of folders is exposed to the user.
- Are there any existing standards or RFCs we could utilise here?

I consider this a low priority/nice to have feature, but I find it very useful and I wanted to mention it. 

What do you think?

**Comment by Fell** [comment]
Thank you for consideration. I fully understand there are probably more important things right now. 

Thinking about it again: Mox already executes "special behavior" when messages are moved to the Junk folder, doesn't it? It shouldn't bee too hard to implement, then. I might have some more time early next year, so I may have a *Go* at it then.

**Comment by mjl-** [comment]
> Thinking about it again: Mox already executes "special behavior" when messages are moved to the Junk folder, doesn't it?

Correct. And also for a few other mailboxes (configurable). See https://github.com/search?q=repo%3Amjl-%2Fmox%20JunkFlagsForMailbox&type=code. The difference is that the snooze functionality requires a timer, to make changes later on. The junk/notjunk flags action can be taken immediately. But indeed the snooze functionality may have to be triggered at the same kind of places (in the code).
